src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.GssApi.cs

```diff
@@ -0,0 +1,125 @@
+
+            using (outputToken)
+            {
+                if ((status != NetSecurityNative.Status.GSS_S_COMPLETE) && (status != NetSecurityNative.Status.GSS_S_CONTINUE_NEEDED))
+                {
+                    throw new NetSecurityNative.GssApiException(SR.net_context_establishment_failed, status, minorStatus);
+                }
+
+                outputBuffer = new byte[outputLength];
+                outputToken.Copy(outputBuffer, 0);
+            }
+            return status == NetSecurityNative.Status.GSS_S_COMPLETE;
+        }
+
+        internal static byte[] Encrypt(
+            SafeGssContextHandle context,
+            bool encrypt,
+            byte[] buffer,
+            int offset,
+            int count)
+        {
+            Debug.Assert((buffer != null) && (buffer.Length > 0), "Invalid input buffer passed to Encrypt");
+            Debug.Assert((offset >= 0) && (offset < buffer.Length), "Invalid input offset passed to Encrypt");
+            Debug.Assert((count > 0) && (count <= (buffer.Length - offset)), "Invalid input count passed to Encrypt");
+
+            SafeGssBufferHandle outputToken;
+            int msgLength;
+            NetSecurityNative.Status minorStatus;
+            NetSecurityNative.Status status = NetSecurityNative.Wrap(out minorStatus, context, encrypt, buffer, offset, count, out outputToken, out msgLength);
+            using (outputToken)
+            {
+                if (status != NetSecurityNative.Status.GSS_S_COMPLETE)
+                {
+                    throw new NetSecurityNative.GssApiException(SR.net_context_wrap_failed, status, minorStatus);
+                }
+
+                byte[] outputBuffer = new byte[msgLength];
+                outputToken.Copy(outputBuffer, 0);
+                return outputBuffer;
+            }
+        }
+
+        internal static int Decrypt(
+            SafeGssContextHandle context,
+            byte[] buffer,
+            int offset,
+            int count)
+        {
+            Debug.Assert((buffer != null) && (buffer.Length > 0), "Invalid input buffer passed to Decrypt");
+            Debug.Assert((offset >= 0) && (offset < buffer.Length), "Invalid input offset passed to Decrypt");
+            Debug.Assert((count > 0) && (count <= (buffer.Length - offset)), "Invalid input count passed to Decrypt");
+
+            SafeGssBufferHandle outputToken;
+            int msgLength;
+            NetSecurityNative.Status minorStatus;
+            NetSecurityNative.Status status = NetSecurityNative.Unwrap(out minorStatus, context, buffer, offset, count, out outputToken, out msgLength);
+
+            using (outputToken)
+            {
+                if (status != NetSecurityNative.Status.GSS_S_COMPLETE)
+                {
+                    throw new NetSecurityNative.GssApiException(SR.net_context_unwrap_failed, status, minorStatus);
+                }
+
+                int length = msgLength - offset;
+                if (msgLength > length)
```

Comment:
This looks odd.  This is essentially checking ```if (msgLength > (msgLength - offset))``` which  is just checking ```if (offset > 0)```.  Why is a positive offset invalid?  Isn't that the point of an offset?